src/hooks/useApproveAndDeposit.ts

```diff
@@ -61,17 +62,19 @@ export function useApproveAndDeposit(
     infiniteApproval,
   } = useSelector((state: AppState) => state.user)
   const POOL = POOLS_MAP[poolName]
+  const basicPools = useContext(BasicPoolsContext)
+  const pool = basicPools?.[poolName]
   const metaSwapContract = useMemo(() => {
-    if (POOL.metaSwapAddresses && chainId && library) {
+    if (pool?.metaSwapDepositAddress && chainId && library) {
       return getContract(
-        POOL.metaSwapAddresses?.[chainId],
+        pool.metaSwapDepositAddress[chainId],
```

Comment:
@hammeiam can you confirm this naming conversion is correct? I read the comment about it but want to make sure.